exercises/subprograms/challenge_121.py

```diff
@@ -0,0 +1,122 @@
+"""
+Create a program that will allow the user to easily manage a list of names.
+You should display a menu that will allow them to add a name to the list,
+change a name in the list, delete a name from the list or view all the names
+in the list.
+There should also be a menu option to allow the user to end the program.
+If they select an option that is not relevant, then it should display a suitable
+message. After they have made a selection to either add a name, change a name,
+delete a name or view all the names, they should see the menu again without having
+to restart the program. The program should be made as easy to use as possible.
+
+"""
+# used sys to terminate the progran execution
+import sys
+
+
+def get_user_input_integer(message):
+    while True:
+        user_input = input(message)
+        if user_input.isdigit():
+            return int(user_input)
+        print("Oops! That was not a valid number. Try again...")
+
+
+def get_user_input_string(message):
+    while True:
+        user_input = input(message)
+        if len(user_input) > 0:
+            return user_input
+        print("Oops! That was not an empty string. Try again...")
+
+
+def ask_for_option_within_list(names_list):
+    while True:
+        display_list(names_list)
+        selected_index = get_user_input_integer(
+            "Enter the number of the name you want to change: "
+        )
+        if 0 < selected_index <= len(names_list):
+            return selected_index - 1
+
+        print("Oops! Please select a number within the list. Try again...")
```

Comment:
Same thing as in challenge 119, can you change the flow so that this message prints as a result of a specific test?

Reply:
Updated